src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs

```diff
@@ -391,6 +391,9 @@ public static string GetTypeInfoName(ITypeInfo typeInfo)
         [MethodImpl(MethodImplOptions.InternalCall)]
         public static extern object GetUniqueObjectForIUnknown(IntPtr unknown);
 
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        public static extern object GetUniqueObjectForIUnknownIgnoreWinRT(IntPtr unknown);
```

Comment:
Would `System.Runtime.InteropServices.WindowsRuntime.WindowsRuntimeMarshal` be better place for this?

Reply:
Also, would calling this `GetUniqueObjectForIUnknownWithoutUnboxing` better capture what it does? (Matches ` InterfaceMarshaler.ConvertToManagedWithoutUnboxing`).

Reply:
I think renaming it for consistency is worth it. Although I think if I do that, I'll keep it in the Marshal class for consistency with the rest of the `Get*ObjectForIUnknown` family of methods.

Reply:
Note that `PtrToStringHString` is on `WindowsRuntimeMarshal`, even though `PtrToStringUni` and other non-WinRT specific APIs are on `Marshal`.

I do not think that the consistency you are shooting for exists today, and we are actually intentionally inconsistent.

Reply:
Didn't notice that. I'll move it over

Reply:
There are a few functions (primarily FCalls) on Marshal that are WinRT-focused, such as `InitializeWrapperForWinRT`, `InitializeManagedWinRTFactoryObject`, and `GetHRForException_WinRT`. It looks like they're publically exposed on `WindowsRuntimeMarshal` but the FCalls are implemented on the `Marshal` class. I can follow that pattern (keep the FCalls on Marshal but publicly expose on WindowsRuntimeMarshal) if you would prefer.

Reply:
Yes, that would be better.

It would be even better for this and other WinRT-specific FCalls directly from WindowsRuntimeMarshal.

Reply:
I'll move the others to be wired through WindowsRuntimeMarshal